pkg/datapath/linux/node.go

```diff
@@ -250,47 +250,51 @@ func installDirectRoute(CIDR *cidr.CIDR, nodeIP net.IP) (routeSpec *netlink.Rout
 	return
 }
 
-func (n *linuxNodeHandler) updateDirectRoute(oldCIDR, newCIDR *cidr.CIDR, oldIP, newIP net.IP, firstAddition, directRouteEnabled bool) error {
+func (n *linuxNodeHandler) updateDirectRoutes(oldCIDRs, newCIDRs []*cidr.CIDR, oldIP, newIP net.IP, firstAddition, directRouteEnabled bool) error {
```

Comment:
I reviewed the logic in this PR and it LGTM in terms of logical gaps from the old implementation. But to be really sure, I noticed that we are lacking some unit tests for this specific function so it might be worth writing unit tests against the old version first, then apply this commit and see if anything breaks (of course there will be slight modifications required as the func params changed, etc).

Reply:
Thanks! That's a really good suggestion. Will implement asap.

Reply:
Turns out, we already have a pretty good unit test for the function: https://github.com/cilium/cilium/blob/6237e74b4f0ffbfc38b5c26f5a0a8f94f33a6c69/pkg/datapath/linux/node_linux_test.go#L646

Doesn't seem like it regressed (run-time tests passed before). But I'll extend it to make sure we cover all corner cases and add support the new behavior.

Reply:
Extended the unit tests accordingly: https://github.com/cilium/cilium/pull/18887/commits/d646e1a17356d08f14e57e48da10646187510f9f